### PR TITLE
Adds new events class into frontend toolkit.

### PR DESCRIPTION
**Pull request by DilwoarH**
This class allows users to easily set google event tracking on buttons and links across the Digital marketplace.

This has been copied over from the buyer frontend with an additional new feature to track a specific input or selection value.

Ticket: https://trello.com/c/FMuTKjo4/114-analytics-for-new-award-flow

### CODE REVIEW

**Comment by katstevens** [comment]
We have some unit tests for `trackEvent` in the Buyer FE - is it possible to transfer these over as well?

File: toolkit/javascripts/analytics/_events.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      trackEvent: function (e) {
+        var $target = $(e.target);
+        var category = $target.attr('data-analytics-category');
+        var action = $target.attr('data-analytics-action');
+        var label = $target.attr('data-analytics-label');
+        
+        // if data-analytics-target-selector is set
+        // then get the value of the target
+        // selector needs to be a CSS selector
+        // ie. ".className", "#id", "input[name=NameOfRadio]:checked"
```
Comment (galund) [inline toolkit/javascripts/analytics/_events.js]: Pedantry alert: "e.g." not "ie"

File: toolkit/templates/documents.html [hunk 2]
```diff
@@ -7,6 +7,7 @@
         {% if item.analytics_category %}data-analytics-category="{{ item.analytics_category }}" {% endif %}
         {% if item.analytics_action %}data-analytics-action="{{ item.analytics_action }}" {% endif %}
         {% if item.analytics_label %}data-analytics-label="{{ item.analytics_label }}" {% endif %}
+        {% if item.analytics_target %} data-analytics-target-selector="{{ item.analytics_target }}" {% endif %}
```
Comment (galund) [inline toolkit/templates/documents.html]: You have acquired an extra space that is going to annoy those of us with OCD tendencies :)

File: toolkit/templates/summary-table.html [hunk 3]
```diff
@@ -166,6 +167,7 @@ <h2 class="summary-item-heading"{% if id %} id="{{ id }}"{% endif %}>
           {% if analytics_category %}data-analytics-category="{{ analytics_category }}" {% endif %}
           {% if analytics_action %}data-analytics-action="{{ analytics_action }}" {% endif %}
           {% if analytics_label %}data-analytics-label="{{ analytics_label }}" {% endif %}
+          {% if analytics_target %} data-analytics-target-selector="{{ analytics_target }}" {% endif %}
```
Comment (galund) [inline toolkit/templates/summary-table.html]: ditto spacing

**Comment by DilwoarH** [comment]
hey @katstevens - doesn't seem any tests exist for the analytics stuff in the toolkit. I think it would be better to do that as a separate ticket as this is blocking some features going live for our mission. What do you think?

**Comment by katstevens** [comment]
@DilwoarH That was my point 😸 There aren't any in the toolkit yet but we could potentially port over the ones from the buyer FE along with the code we've lifted from there. The tech debt ticket (https://trello.com/c/vgVAIG4n/233-shared-analytics-js-in-frontend-toolkit) has been lingering for ages because I wasn't sure how to get those tests working in the toolkit - I was hoping you'd have some ideas! If it's not straightforward then it's fine to leave it out of this PR.

**Pull request merged**